### PR TITLE
Updated docs, removed Inject annotation, and small code fixes

### PR DESCRIPTION
**Pull request by szczygiel-m**

### CODE REVIEW

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/di/factories/MetricRegistryFactory.java [hunk 1]
```diff
@@ -34,15 +32,14 @@
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 
-public class MetricRegistryFactory implements Factory<MetricRegistry> {
+public class MetricRegistryFactory {
```
Comment (piotrrzysko) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/di/factories/MetricRegistryFactory.java:35]: I would get rid of these factories. In my opinion, they duplicate that we do in `*Configuration` classes.

File: docs/docs/configuration/message-tracking.md [hunk 2]
```diff
@@ -31,31 +31,31 @@ in Frontend, Consumers and Management.
 
 ### Frontend configuration
 
-* create `ElasicsearchClientFactory`, which will produce ElasticSearch driver and store it for cleanup
-* set log repository via `HermesFrontend.Builder#withLogRepository`
-* add shutdown hook to cleanup the driver
```
Comment (piotrrzysko) [inline docs/docs/configuration/message-tracking.md:36]: It is not directly related to this line, but the line reminded me that in the consumers module I added the shutdown hook here: https://github.com/allegro/hermes/blob/cf78bb22a936fe9326bc3d00db851d42d1976d0b/hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/ConsumerConfiguration.java#L221 Probably we should do the same in the frontend module.

File: docs/docs/configuration/buffer-persistence.md [hunk 3]
```diff
@@ -56,16 +56,25 @@ class BrokerListener implements BrokerAcknowledgedListener,
         /* ... */
     }
 }
+```
 
-class HermesStarter {
+```java
+@Configuration
+public class CustomHermesFrontendConfiguration {
```
[the remark this answers is not in the document]
Comment (szczygiel-m) [reply docs/docs/configuration/buffer-persistence.md:63]: good point, changed to: 
`@Configuration
public class CustomHermesFrontendConfiguration {

    @Bean(initMethod = "extend")
    public PersistentBufferExtension persistentBufferExtension(ConfigFactory configFactory,
                                                               Clock clock,
                                                               BrokerListeners listeners,
                                                               BackupMessagesLoader backupMessagesLoader,
                                                               HermesMetrics hermesMetrics) {
        BrokerListener customBrokerListener = new MyCustomBrokerListener();
        BrokerListeners brokerListeners = new BrokerListeners();
        
        brokerListeners.addAcknowledgeListener(customBrokerListener);
        brokerListeners.addTimeoutListener(customBrokerListener);
        brokerListeners.addErrorListener(customBrokerListener);
        
        return new PersistentBufferExtension(configFactory, clock, brokerListeners, backupMessagesLoader,
                hermesMetrics);
    }
}`

[the remark this answers is not in the document]
Comment (piotrrzysko) [reply docs/docs/configuration/buffer-persistence.md:63]: Last comment :) 

Wouldn't be easier for a user to do something like this:
```java
@Primary
@Bean
public BrokerListeners myBrokerListeners() {
    BrokerListener customBrokerListener = new MyCustomBrokerListener();
    BrokerListeners brokerListeners = new BrokerListeners();
    
    brokerListeners.addAcknowledgeListener(customBrokerListener);
    brokerListeners.addTimeoutListener(customBrokerListener);
    brokerListeners.addErrorListener(customBrokerListener);
    
    return brokerListeners;
}
```

Then they don't need to worry about creating and initializing an instance of `PersistentBufferExtension`, which is pretty complicated.

**Review by piotrrzysko — APPROVED** [review]

**Pull request merged**